src/Symfony/Component/String/Inflector/FrenchInflector.php

```diff
@@ -0,0 +1,257 @@
+        'anchois',
+        'anglais',
+        'autobus',
+        'brebis',
+        'carquois',
+        'cas',
+        'chas',
+        'colis',
+        'concours',
+        'corps',
+        'cours',
+        'cyprès',
+        'décès',
+        'devis',
+        'discours',
+        'dos',
+        'embarras',
+        'engrais',
+        'entrelacs',
+        'excès',
+        'fois',
+//        'fonds',
+        'gâchis',
+        'gars',
+        'glas',
+//        'guet-apens',
+        'héros',
+        'intrus',
+        'jars',
+        'jus',
+        'kermès',
+        'lacis',
+        'legs',
+        'lilas',
+        'marais',
+        'mars',
+        'matelas',
+        'mépris',
+        'mets',
+        'mois',
+        'mors',
+        'obus',
+        'os',
+        'palais',
+        'paradis',
+        'parcours',
+        'pardessus',
+        'pays',
+        'plusieurs',
+        'poids',
+        'pois',
+        'pouls',
+        'printemps',
+        'processus',
+        'progrès',
+        'puits',
+        'pus',
+        'rabais',
+        'radis',
+        'recors',
+        'recours',
+        'refus',
+        'relais',
+        'remords',
+        'remous',
+        'rhinocéros',
+        'repas',
+        'rubis',
+        'sas',
+        'secours',
+        'souris',
+        'succès',
+        'talus',
+        'tapis',
+        'taudis',
+        'temps',
+        'tiers',
+        'univers',
+        'velours',
+        'verglas',
+        'vernis',
+        'virus',
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function singularize(string $plural): array
+    {
+        //Inflected words
+        if ($this->isInflectedWord($plural)) {
+            return [$plural];
+        }
+
+        //Exceptions rules
+        foreach (self::$singularizeRegexp as $rule) {
+            list($regexp, $replace) = $rule;
+
+            if (1 === preg_match($regexp, $plural)) {
+                return [preg_replace($regexp, $replace, $plural)];
+            }
+        }
+
+        // Default rule
+        return [$plural];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function pluralize(string $singular): array
+    {
+        //Inflected words
+        if ($this->isInflectedWord($singular)) {
+            return [$singular];
+        }
+
+        foreach (self::$pluralizeRegexp as $rule) {
+            list($regexp, $replace) = $rule;
+
+            if (1 === preg_match($regexp, $singular)) {
+                return [preg_replace($regexp, $replace, $singular)];
+            }
+        }
+
+        // Default rule
+        return [$singular.'s'];
+    }
+
+    /**
+     * Is this word an inflected one?
+     *
+     * This method constructs a regexp by imploding all value from uninflected array
+     *
+     * @param string $word word to test
+     */
+    private function isInflectedWord(string $word): bool
+    {
+        $regexp = '/^('.implode('|', self::$uninflected).')$/i';
```

Comment:
should we keep the result in a static property since it's a static string?

Reply:
I saw two solutions:

1. Static singleton: I declare a private static regexp as null, I add a private method to get it. If it is null, I calculate it.
2. Static uninflected variable, can be the result string but the line where it will be declared will be very long (569 chars in the string).
Solution1:
```
    private function isInflectedWord(string $word): bool
    {
        if (null === self::$regexp) {
            self::$regexp = '/^('.implode('|', self::$uninflected).')$/i';
        }

        return 1 === preg_match(self::$regexp, $word);
    }
```
Solution2:
`private static $uninflected = '/^(abcès|accès|abus|albatros|anchois|anglais|autobus|brebis|carquois|cas|chas|colis|concours|corps|cours|cyprès|décès|devis|discours|dos|embarras|engrais|entrelacs|excès|fois|gâchis|gars|glas|héros|intrus|jars|jus|kermès|lacis|legs|lilas|marais|mars|matelas|mépris|mets|mois|mors|obus|os|palais|paradis|parcours|pardessus|pays|plusieurs|poids|pois|pouls|printemps|processus|progrès|puits|pus|rabais|radis|recors|recours|refus|relais|remords|remous|rhinocéros|repas|rubis|sas|secours|souris|succès|talus|tapis|taudis|temps|tiers|univers|velours|verglas|vernis|virus)$/i'`

Reply:
I finally use solution2